cypress/GenerateCypressTestSpecs.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+*/
+
+// ----------------------------------------------------------------------
+// This code is intended to be run via NODE.js.
+// It is intended to be started automatically whenever test cases 
+// are being initialized to run by Cypress.
+//
+// This code parses all .js files in the cypress/tests folder.
+// It is looking for lines in the files that look like this:
+//  Cypress.TestCase(testGroupName, testDescription, testFunction)
+//
+// It uses that data to create:
+//   1) The group subfolders within "cypress/integration".
+//   2) An individual test spec file for each, in that same folder as 
+//      expected by Cypress.
+//   3) The test list: masterListOfAllTestCases
+// ----------------------------------------------------------------------
+
+var fs = require('../node_modules/pn/fs')
```

Comment:
This is risky and should be avoided. It seems your digging into the `node_modules` folder and assuming `pn` will be there, but it may not.  If any of dependencies decide to use a different package we would break. We're only supposed to depend on packages we explicitly declare.

```
npm ls pn
@conversationlearner/ui@0.0.0-development D:\repos\cl-ui
`-- react-scripts-ts@3.1.0
  `-- ts-jest@22.0.1
    `-- jest-config@22.4.4
      `-- jest-environment-jsdom@22.4.3
        `-- jsdom@11.12.0
          `-- pn@1.1.0`
```

At the very least I wonder if `var fs = require('pn/fs');` works?

Reply:
That one I just now fixed. It was an artifact of trying to get a version of this code running under Cypress.